### ISSUE TITLE
About the setting of experiments

### ISSUE THREAD

**Issue by daidaiershidi** [EnnengYang/SurgeryV2#1]
Thank you for the great work. I'd like to understand what is the definition of model merge? You seem train a set of parameters for each task. Shouldn't model merge use the same set of parameters for all tasks?

**Comment by EnnengYang** [comment]
Hi,

Thank you for your attention to our work.

In fact, in the “**standard**” definition of model merging[1,2], all tasks share exactly the same parameters.

However, complete parameter sharing may limit the performance of the merged model. Some recent studies have also tried to set private masks for tasks[3] or merge model parameters dynamically for each task/sample[4,5] instead of fully sharing parameters. However, these methods have additional costs while improving performance, such as the need to train routing/extra storage costs. Similarly, our Surgery also has some potential limitations, such as the need to tune the Surgery modules and the need for task IDs to select the appropriate route (although there are many strategies to solve this, such as referring to the dynamic merging method to add a trainable route to identify the task, or designing a training-free route based on the idea of ​​Prototypical Networks[6]). These are also directions that can be further optimized in our future work.

In summary, fully sharing parameters may limit performance, and partially sharing parameters may rely on task identification or additional training/storage burdens. Current model merging methods need to carefully balance the two.

In particular, from my perspective, the **ideal model merging method** should be: parameter efficient (for example, all tasks fully share parameters), training free (direct merging without relying on additional optimization or search), high performance (having the same capabilities as expert models on downstream tasks), good generalization (the merged model helps solve new tasks), good stability (the general capabilities of the pre-trained model will not be forgotten), etc. However, these goals are still challenging and require further efforts from the community.

References:
[1] Wortsman, Mitchell, et al. "Model soups: averaging weights of multiple fine-tuned models improves accuracy without increasing inference time." International conference on machine learning. PMLR, 2022.
[2] Ilharco, Gabriel, et al. "Editing models with task arithmetic." The Eleventh International Conference on Learning Representations, 2023.
[3] Huang, Chenyu, et al. "EMR-Merging: Tuning-Free High-Performance Model Merging." The Thirty-eighth Annual Conference on Neural Information Processing Systems. 2024.
[4] Lu, Zhenyi, et al. "Twin-Merging: Dynamic Integration of Modular Expertise in Model Merging." The Thirty-eighth Annual Conference on Neural Information Processing Systems. 2024.
[5] Tang, Anke, et al. "Merging Multi-Task Models via Weight-Ensembling Mixture of Experts." Forty-first International Conference on Machine Learning. 2024.
[6] Snell, Jake, Kevin Swersky, and Richard Zemel. "Prototypical networks for few-shot learning." Advances in neural information processing systems 30 (2017).


Best,
Enneng
